src/test/java/com/github/dockerjava/cmd/PullImageCmdIT.java

```diff
@@ -92,25 +93,19 @@ public void testPullImage() throws Exception {
         assertThat(inspectImageResponse, notNullValue());
     }
 
-
-
     @Test
     public void testPullNonExistingImage() throws Exception {
-        // does not throw an exception
-
-        if (getVersion(dockerRule.getClient()).equals(RemoteApiVersion.VERSION_1_24)) {
-            // 1.24 - no exception
-        } else if (getVersion(dockerRule.getClient())
-                .isGreaterOrEqual(RemoteApiVersion.VERSION_1_26)) {
-            exception.expect(NotFoundException.class);
-        } else {
-            exception.expect(DockerClientException.class);
+        //different docker version throws different errors here
+        try {
+            // stream needs to be fully read in order to close the underlying connection
+            dockerRule.getClient().pullImageCmd("xvxcv/foo")
+                    .exec(new PullImageResultCallback())
+                    .awaitCompletion(30, TimeUnit.SECONDS);
+
+            fail("An exception is expected.");
+        } catch (NotFoundException | DockerClientException e) {
```

Comment:
that a wrote junit exception usage. test shouldn't supress exceptions, it need only configure exception (that is Rule).

Reply:
https://github.com/junit-team/junit4/wiki/Exception-testing#expectedexception-rule

Reply:
But thats an or exception condition
maybe we let DockerClientException extend DockerException?

Reply:
no idea, how it influence on this test case? Different docker versions return different error codes so client returns different exceptions. That's why Exception Rule was conditionally configured before call that should throw error.

Reply:
It will change that each method will at least throw a DockerException. So you can always expect a DockerException regardless which docker version you exactly use.
Currently you must expect a DockerException or a DockerClientException.